hook.sh

```diff
@@ -135,17 +144,38 @@ function clean_challenge {
     fi
 
     test -f transaction.yaml && rm transaction.yaml
-    gcloud dns record-sets transaction start --zone $zonename
-
-    existingRecord=`gcloud dns record-sets list --name "_acme-challenge.$DOMAIN." --type TXT --zone $zonename --format='value(name,rrdatas[0],ttl)'`
-    existingRecord=${existingRecord//$'\t'/,}
-    IFS=$',' read existingName existingRrdata existingTtl <<< "$existingRecord"
 
-    # Replace threefold """ with singe "
-    existingRrdata=${existingRrdata//$'"""'/''}
-
-    gcloud dns record-sets transaction remove --name $existingName --type TXT --ttl $existingTtl --zone $zonename -- "$existingRrdata"
-    gcloud dns record-sets transaction execute --zone $zonename
+    existingRecord=`gcloud dns record-sets list --name "_acme-challenge.$DOMAIN." --type TXT --zone $zonename --format='value(name,rrdatas,ttl)'`
+    IFS=$'\t' read existingName existingRrdata existingTtl <<< "$existingRecord"
+    existingRrdata=${existingRrdata//$'"'/''}
+    IFS=',' read -ra existingRrdata <<< "$existingRrdata"
+
+    if [[ ${#existingRrdata[@]} > 1 ]]; then
+      # Remove all record
+      gcloud dns record-sets transaction start --zone $zonename
+      stringify=${existingRrdata[@]}
+      stringify=${stringify// /"\" \""}
```

Comment:
This is adding another tool requirement to the script - is there no way we can do that in bash? Also, please name variables according to what is in there instead of naming them after the command you got the data from.